spec/models/manageiq/providers/hawkular/middleware_manager/alert_manager_spec.rb

```diff
@@ -0,0 +1,171 @@
+describe ManageIQ::Providers::Hawkular::MiddlewareManager::AlertManager do
+  let(:client) { double('Hawkular::Alerts') }
+  let(:stubbed_ems) do
+    ems = instance_double('::ManageIQ::Providers::Hawkular::MiddlewareManager',
+                          :alerts_client => client,
+                          :id            => 5)
+    allow(ems).to receive(:miq_id_prefix) { |id| id }
+    ems
+  end
+  let(:subject) { described_class.new(stubbed_ems) }
+
+  it '#convert_operator' do
+    expect(subject.send(:convert_operator, "<")).to eq(:LT)
+    expect(subject.send(:convert_operator, "<=")).to eq(:LTE)
+    expect(subject.send(:convert_operator, ">=")).to eq(:GTE)
+    expect(subject.send(:convert_operator, "=")).to eq(:LTE)
+    expect(subject.send(:convert_operator, ">")).to eq(:GT)
+    expect(subject.send(:convert_operator, "wrong operator")).to eq(nil)
+  end
+
+  it '#generate_mw_threshold_condition' do
+    c = subject.send(:generate_mw_threshold_condition, "1", :LT, 3)
+    expect(c).to be_a_kind_of(Hawkular::Alerts::Trigger::Condition)
```

Comment:
As Condition is a generic object, I think it could be good to test that type == THRESHOLD, as there are several types of conditions available.

Reply:
Also this comments can be extended to rest of specs

Reply:
I did the changes, thanks @lucasponce